scio-elasticsearch/es7/src/main/scala/com/spotify/scio/elasticsearch/ElasticsearchIO.scala

```diff
@@ -96,7 +100,8 @@ object ElasticsearchIO {
     numOfShards: Long = WriteParam.DefaultNumShards,
     maxBulkRequestSize: Int = WriteParam.DefaultMaxBulkRequestSize,
     maxBulkRequestBytes: Long = WriteParam.DefaultMaxBulkRequestBytes,
-    retry: RetryConfig = WriteParam.DefaultRetryConfig
+    retry: RetryConfig = WriteParam.DefaultRetryConfig,
+    usernameAndPassword: Option[(String, String)] = None
```

Comment:
I was thinking that we could add these in `ElasticsearchOptions`. It kinda is like a `ConnectionOptions`. (which is a far better name)

Reply:
Created and issue around the naming #3573

Reply:
Is it allowed to introduce breaking changes?

Reply:
I would rather not introduce breaking changes at least with this PR. From a brief look i think it's possible to easily deprecate that type.

Reply:
I'd like to add this overloaded method like below not to introduce breaking changes.
```scala
def saveAsElasticsearch(
      connectionOptions: ConnectionOptions)(
      flushInterval: Duration = WriteParam.DefaultFlushInterval,
      numOfShards: Long = WriteParam.DefaultNumShards,
      maxBulkRequestSize: Int = WriteParam.DefaultMaxBulkRequestSize,
      maxBulkRequestBytes: Long = WriteParam.DefaultMaxBulkRequestBytes,
      errorFn: BulkExecutionException => Unit = WriteParam.DefaultErrorFn,
      retry: RetryConfig = WriteParam.DefaultRetryConfig
    )(f: T => Iterable[DocWriteRequest[_]]): ClosedTap[Nothing]
```

But it can't be compiled due to `in class ElasticsearchSCollection, multiple overloaded alternatives of method saveAsElasticsearch define default arguments`. Is there any reasonable method not to introduce breaking changes? 

And if we change the option name to `ConnectionOptions`, users might think it's natural that ConnectionOptions has RetryConfig and maxBulkRequestXXX. For me, keeping name as ElasticsearchOptions is not so bad.

Reply:
>But it can't be compiled due to in class ElasticsearchSCollection, multiple overloaded alternatives of method saveAsElasticsearch define default arguments. Is there any reasonable method not to introduce breaking changes?

That's the downside of using default arguments, unfortunately. We would have to add a new method.

Let's do the following let's explore the ElasticsearchOptions vs ConnectionOptions on a follow-up PR and just wrap this up with moving username/password to ConnectionOptions. wdyt?

Reply:
> Let's do the following let's explore the ElasticsearchOptions vs ConnectionOptions on a follow-up PR and just wrap this up with moving username/password to ConnectionOptions. wdyt?

Sounds good!